ci.sh

```diff
@@ -174,26 +174,32 @@ if [ "$PROVIDER_NAME" = "trusted-service" ]; then
 fi
 
 if [ "$PROVIDER_NAME" = "coverage" ]; then
-    PROVIDERS="mbed-crypto tpm pkcs11"
-    # Install tarpaulin and start Parsec with it
+    PROVIDERS="mbed-crypto tpm" # pkcs11 not supported because of a segfault when the service stops
+    EXCLUDES="fuzz/*,e2e_tests/*,src/providers/cryptoauthlib/*,src/providers/pkcs11/*,src/providers/trusted_service/*"
+    # Install tarpaulin
     cargo install cargo-tarpaulin
 
     mkdir -p reports
 
     for provider in $PROVIDERS; do
+        # Set up run
         PROVIDER_NAME=$provider
         TEST_FEATURES="--features=$provider-provider"
         cp $(pwd)/e2e_tests/provider_cfg/$provider/config.toml $CONFIG_PATH
         mkdir -p reports/$provider
-        RUST_LOG=info cargo tarpaulin --out Xml --forward --command build --output-dir $(pwd)/reports/$provider --features="all-providers,all-authenticators" --run-types bins --timeout 3600 -- -c $CONFIG_PATH &
+
+        # Start service
+        RUST_LOG=info cargo tarpaulin --out Xml --forward --command build --exclude-files="$EXCLUDES" --output-dir $(pwd)/reports/$provider --features="all-providers,all-authenticators" --run-types bins --timeout 3600 -- -c $CONFIG_PATH &
```

Comment:
Actually, does it make sense to do `all-providers` then? Should we remove the one we ignore? Are they automatically ignored if they are not compiled in?

Reply:
I just exclude them through the CLI option and they don't get instantiated anyway. Could change it to only compile what we need!